### PR TITLE
nautilus: mgr/dashboard: Hardening accessing the metadata

### PR DESCRIPTION
**Pull request by votdev**
Partly backported from https://tracker.ceph.com/issues/41372. This is done because we only want to backport the hardened code to access the metadata information.
 Fixes: https://tracker.ceph.com/issues/43079
 Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

**Comment by Devp00l** [comment]
Please backport also the [unit tests](https://github.com/ceph/ceph/pull/29800/files#diff-8f1095d0d1a854804fcbf82bb37c9466) for this problem. They can be found at the bottom of my merged PR.

File: src/pybind/mgr/dashboard/tests/test_rbd_mirroring.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class RbdMirroringSummaryControllerTest(ControllerTestCase):
     @classmethod
     def setup_server(cls):
         mgr.list_servers.return_value = mock_list_servers
-        mgr.get_metadata.return_value = mock_get_metadata
+        mgr.get_metadata = mock.Mock(return_value=mock_get_metadata)
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard/tests/test_rbd_mirroring.py]: Is this change needed? As it seems unrelated.

Comment (votdev) [reply src/pybind/mgr/dashboard/tests/test_rbd_mirroring.py]: > Is this change needed?

I don't think so. But i looked more correct to me, so i added it. Will remove it again.

Comment (votdev) [reply src/pybind/mgr/dashboard/tests/test_rbd_mirroring.py]: Had to add it again, otherwise the unit tests will fail.

**Comment by smithfarm** [comment]
@vtheile Over time, we developed a set of cherry-picking rules [1] or principles. One of these generally accepted principles is, when cherry-picking, leave the original commit message intact. 

Would you mind re-doing the cherry-pick and move your comments below the "(cherry picked from commit ...)" line, and leave everything above that line untouched?

**Comment by votdev** [comment]
> @vtheile Over time, we developed a set of cherry-picking rules [1] or principles. One of these generally accepted principles is, when cherry-picking, leave the original commit message intact.
> 
> Would you mind re-doing the cherry-pick and move your comments below the "(cherry picked from commit ...)" line, and leave everything above that line untouched?

I fear i can not do that because this PR is absolutely no 1:1 cherry-pick, it has 0% relation to the PR which is cherry-picked, it only contains the code of interest. Because of that it does not make sense to leave the original PR description. I also could have implemented this bugfix PR from scratch without mentioning the PR where the code comes from.

**Comment by smithfarm** [comment]
@votdev In that case, could you remove the line `(cherry picked from commit ...)` from the commit and replace it with a sentence explaining why the commit could not be cherry picked from master?

**Comment by votdev** [comment]
> @votdev In that case, could you remove the line `(cherry picked from commit ...)` from the commit?
> ~and replace it with a sentence explaining why the commit could not be cherry picked from master?~
> 
> (actually, you already explain that more or less, so just need to drop the line that indicates it's a cherry-pick)

Done.

**Comment by yuriw** [comment]
[wip-yuri2-testing-2020-01-15-2019-nautilus](https://trello.com/c/0kpGomOf/808-wip-yuri2-testing-2020-01-15-2019-nautilus)

**Comment by yuriw** [comment]
retest this please
